fe/fe-core/src/main/java/org/apache/doris/analysis/IndexDef.java

```diff
@@ -142,6 +217,30 @@ public void checkColumn(Column column, KeysType keysType) throws AnalysisExcepti
                         "BITMAP index only used in columns of DUP_KEYS/UNIQUE_KEYS table or key columns of"
                                 + " AGG_KEYS table. invalid column: " + indexColName);
             }
+        } else if (indexType == IndexType.NGRAM_BF) {
+            String indexColName = column.getName();
+            PrimitiveType colType = column.getDataType();
+            if (colType != PrimitiveType.CHAR && colType != PrimitiveType.VARCHAR) {
```

Comment:
colType.isStringType() may be better, if String is also supported.

Reply:
you are right. but i will change to colType.isCharFamily(), as isStringType includes hll type.